FireboltNETSDK/Client/FireboltCommand.cs

```diff
@@ -166,7 +176,17 @@ public override int CommandTimeout
             }
 
             var database = Connection?.Database != string.Empty ? Connection?.Database : null;
-            Task<string?> t = Connection!.Client.ExecuteQuery(engineUrl, database, Connection?.AccountId, SetParamList, newCommandText);
+
+            CancellationToken token = CancellationToken.None;
+            if (CommandTimeoutMillis != 0)
```

Comment:
Doesn't this prevent cancellation of commands without timeout? Why do we have this check?

Reply:
Yes, I understand. My question is why do we have this `if`. If I understand correctly, it prevents us from cancelling query with no timeout.

Reply:
Kind of. 0 value of timeout behaviours exactly as "no timeout". If timeout is 0 we pass `CancellationToken.None` to `ExecuteQueryAsync` and does not register event that will cancel this task automatically when timeout is expired. IMHO this is exactly the "no timeout" behaviour.

Reply:
I understand what the code does, but I don't understand the logic here. If the query has no timeout then I have even more reason to be able to cancel it in case it misbehaves. Timeout and cancellation are not mutually exclusive. User might want to cancel a query from another thread regardless of the timeout set on it.
IMO this not only redundant but harmful.

Reply:
All fixed. Please review.